src/main/java/org/truffleruby/core/inlined/InlinedByteSizeNode.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.core.inlined;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.object.DynamicObject;
+import org.truffleruby.RubyContext;
+import org.truffleruby.core.string.StringNodes;
+import org.truffleruby.language.dispatch.RubyCallNodeParameters;
+
+public abstract class InlinedByteSizeNode extends UnaryInlinedOperationNode {
+
+    public InlinedByteSizeNode(RubyContext context, RubyCallNodeParameters callNodeParameters) {
+        super(callNodeParameters, context.getCoreMethods().stringBytesizeAssumpton);
+    }
+
+    @Specialization(guards = "isRubyString(self)", assumptions = "assumptions")
+    int byteSize(DynamicObject self,
```

Comment:
Should this have an access modifier?

Reply:
I'm following the pattern of the other inlined nodes where no modifier is used. Usually we use `protected` to avoid people using them directly, but package private should be fine as well.